utils/land-patterns.stanza

```diff
@@ -9,23 +9,147 @@ defpackage ocdb/land-patterns :
     prefix(pad) => def-
   import ocdb/design-vars
 
+;==============================================================================
+;============================== Helpers =======================================
+;==============================================================================
 ; ====== Argument constraint Helpers ===========================================
+; Ensure that a value is even and positive.
 defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
 ; ====== Reference Label Helpers ===============================================
-public defn ref-label (x:Double, y:Double, a:Anchor) :
-  inside pcb-landpattern :
-    layer(Silkscreen("values", Top)) = Text(">REF", 0.7056, a, loc(x,y))
-    
+; Creates a reference label from pose and anchor
+public defn ref-label (pose:Pose, anchor:Anchor):
+  inside pcb-landpattern:
+    layer(Silkscreen("values", Top)) = Text(">REF", MIN-SILKSCREEN-TEXT-HEIGHT, anchor, pose)
+
+; Creates a reference label at a fixed pose
+public defn ref-label (pose:Pose):
+  ref-label(pose, C)
+
+; DEPRECATED: prefer ref-label(loc(x, y))
+; Creates a reference label at an x,y position and anchor
+public defn ref-label (x:Double, y:Double, anchor:Anchor):
+  ref-label(loc(x, y), anchor)
+
+; DEPRECATED: prefer ref-label(loc(x, y))
+; Creates a reference label at an x,y position with default anchor
 public defn ref-label (x:Double, y:Double) :
   ref-label(x,y,C)
 
+; Create a default reference label
 public defn ref-label () :
-  ref-label(0.0, 0.0)
+  ref-label(loc(0.0, 0.0))
+
+; Helper function to convert an anchor into a point.
+defn normalize-to-point (anchor:Anchor) -> Point:
+  val [v, h] = components(anchor)
+  val dx = match(h):
+    (hr:W) :  1.0
+    (hr:C) :  0.0
+    (hr:E) : -1.0
+  val dy = match(v) :
+    (vt:S) : 1.0
+    (vt:C) : 0.0
+    (vt:N) : 1.0
+  Point(dx, dy)
+
+; TODO move to internals
```

Comment:
Do you want to do in another branch?

Reply:
Eventually, yes. The issue is that we can't make changes to internals without blocking this PR and the work that depends on it until the infrastructure is up and running.